src/modules/UploadModule.ts

```diff
@@ -102,11 +105,12 @@ export default class UploadModule extends VuexModule implements FfaProcessModule
   }
 
   get hash(): string {
-    const metaMaskModule = this.context.rootState.metaMaskModule as MetaMaskModule
     const web3Module = this.context.rootState.web3Module as Web3Module
+    const metaMaskModule = this.context.rootState.metaMaskModule as MetaMaskModule
+    const flashesModule = this.context.rootState.web3Module as FlashesModule
 
-    if (metaMaskModule.publicKey.length === 0) {
-      throw new Error(Errors.PUBLIC_KEY_EMPTY)
+    if (typeof ethereum === 'undefined' || typeof ethereum.selectedAddress === 'undefined') {
```

Comment:
This isn't working.  Connect metamask and then reload the page.  Metamask is disconnected.

This also violates the same engineering principle (separation of concerns) twice:

1. This method is a getter.  It's role is to define a computed property.  That's all it should ever do.

2. This class is a vuex module.  It should store and retrieve state.  Business logic doesn't belong here.

This block needs to be moved elsewhere where business logic is allowed like `App.vue`.

The logic needs specs (see comment in `Navigation.spec.ts`), the actual connect stuff in a new `App.spec.ts` and the changes in the rendering of the nav in `Navigation.spec.ts`.